editor/src/app/sites/sections/sections-state/site-sections.state.ts

```diff
@@ -133,27 +149,39 @@ export class SiteSectionsState implements NgxsOnInit {
 
   @Action(DeleteSiteSectionAction)
   deleteSiteSection({ getState, setState }: StateContext<SiteSectionStateModel[]>, action: DeleteSiteSectionAction) {
-    const state = getState();
-    let order = -1;
-
-    setState(
-      state
-        .filter(section => !(section.site_name === action.section.site_name && section.name === action.section.name))
-        // Update order
-        .map(section => {
-          if (section.site_name === action.section.site_name) {
-            order++;
-
-            if (section.order !== order) {
-              return {...section, ...{'order': order}};
-            }
-          }
-          return section;
-        })
-    );
+    const data = {
+      site: action.section.site_name,
+      section: action.section.name
+    };
 
-    this.store.dispatch(new DeleteSectionTagsAction(action.section));
-    this.store.dispatch(new DeleteSectionEntriesAction(action.section));
+    this.appStateService.sync('siteSections', data, 'DELETE')
+      .then((response: any) => {
```

Comment:
Let's keep to observables since we're already using them everywhere. So return observable from `sync` (remove the `toPromise`). And add `.pipe(map(response => {...}))` here.